### PR TITLE
Legacy code cleanup

### PR DESCRIPTION
**Pull request by amassarwi**

### CODE REVIEW

**Comment by amassarwi** [comment]
I've found out that the endless loading in account's proposals list was still happening for new users with no proposals, this commit here https://github.com/decred/politeiagui/pull/1828/commits/ea28c461167b1412d707ae24b7ef4e29fff263ea solved the issue only for users with 1 proposal at least.

I've investigated the issue further and found that our condition to determine whether the fetch is done isn't good enough. I changed it to use the redux `user num of proposals`  (on the way fixed it's selector) - see https://github.com/decred/politeiagui/pull/1841/commits/d366011a1025ecb17615fa4873099b6dd33c6328.

File: src/containers/Proposal/User/Submitted.jsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ const Proposals = (props) => {
   }
 
   const numOfProsalsLoaded = proposals.length;
-  const initialFetchDone = !!proposals.length;
+  const initialFetchDone = numOfProsalsLoaded === numOfUserProposals;
```
Comment (tiagoalvesdulce) [inline src/containers/Proposal/User/Submitted.jsx]: Same typo

File: src/containers/Proposal/User/Submitted.jsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ const Proposals = (props) => {
   }
 
   const numOfProsalsLoaded = proposals.length;
```
Comment (tiagoalvesdulce) [inline src/containers/Proposal/User/Submitted.jsx]: There's a typo here. Should be `numOfProposalsLoaded`

**Comment by tiagoalvesdulce** [comment]
For some reason, the order of my comments above got messed up. Everything else looks good. Good job!

**Comment by amassarwi** [comment]
@tiagoalvesdulce thanks mate, updated!

**Pull request merged**